src/features/users/authorSlice.ts

```diff
@@ -0,0 +1,18 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { User } from '../../types/User';
+
+const initialState = null as User | null;
+
+export const authorSlice = createSlice({
+  name: 'author',
+  initialState,
+  reducers: {
+    setAuthor: (_, action: PayloadAction<User>) => {
+      return action.payload;
+    },
```

Comment:
The reducer 'setAuthor' correctly handles the action to set the state. However, it's important to ensure that other parts of the application properly handle the state when it is 'null', especially when logging out a user or when the application starts.